pkg/hintrunner/zero/zerohint_ec.go

```diff
@@ -200,3 +200,166 @@ func newFastEcAddAssignNewYHint() hinter.Hinter {
+			if err != nil {
+				return err
+			}
+			slopeMemoryValues, err := hinter.GetConsecutiveValues(vm, slopeAddr, int16(3))
+			if err != nil {
+				return err
+			}
+
+			point0Addr, err := point0.GetAddress(vm)
+			if err != nil {
+				return err
+			}
+			point0MemoryValues, err := hinter.GetConsecutiveValues(vm, point0Addr, int16(6))
+			if err != nil {
+				return err
+			}
+
+			point1Addr, err := point1.GetAddress(vm)
+			if err != nil {
+				return err
+			}
+			point1MemoryValues, err := hinter.GetConsecutiveValues(vm, point1Addr, int16(3))
+			if err != nil {
+				return err
+			}
+
+			// [d0, d1, d2]
+			var slopeValues [3]*fp.Element
+			// [x.d0, x.d1, x.d2]
+			var point0XValues [3]*fp.Element
+			// [y.d0, y.d1, y.d2]
+			var point0YValues [3]*fp.Element
+			// [x.d0, x.d1, x.d2]
+			var point1XValues [3]*fp.Element
+
+			for i := 0; i < 6; i++ {
+				if i < 3 {
+					slopeValue, err := slopeMemoryValues[i].FieldElement()
+					if err != nil {
+						return err
+					}
+					slopeValues[i] = slopeValue
+
+					point0XValue, err := point0MemoryValues[i].FieldElement()
+					if err != nil {
+						return err
+					}
+					point0XValues[i] = point0XValue
+
+					point1XValue, err := point1MemoryValues[i].FieldElement()
+					if err != nil {
+						return err
+					}
+					point1XValues[i] = point1XValue
+				} else {
+					point0YValue, err := point0MemoryValues[i].FieldElement()
+					if err != nil {
+						return err
+					}
+					point0YValues[i-3] = point0YValue
+				}
+			}
+
+			//> slope = pack(ids.slope, PRIME)
+			slopeBig, err := secp_utils.SecPPacked(slopeValues)
+			if err != nil {
+				return err
+			}
+
+			//> x0 = pack(ids.point0.x, PRIME)
+			x0Big, err := secp_utils.SecPPacked(point0XValues)
+			if err != nil {
+				return err
+			}
+
+			//> x1 = pack(ids.point1.x, PRIME)
+			x1Big, err := secp_utils.SecPPacked(point1XValues)
+			if err != nil {
+				return err
+			}
+
+			//> y0 = pack(ids.point0.y, PRIME)
+			y0Big, err := secp_utils.SecPPacked(point0YValues)
+			if err != nil {
+				return err
+			}
+
+			//> value = new_x = (pow(slope, 2, SECP_P) - x0 - x1) % SECP_P
+
+			secPBig, ok := secp_utils.GetSecPBig()
+			if !ok {
+				return fmt.Errorf("GetSecPBig failed")
+			}
+
+			new_xBig := new(big.Int)
+			new_xBig.Exp(slopeBig, big.NewInt(2), secPBig)
+			new_xBig.Sub(new_xBig, x0Big)
+			new_xBig.Sub(new_xBig, x1Big)
+			new_xBig.Mod(new_xBig, secPBig)
+
+			valueBig := new(big.Int)
+			valueBig.Set(new_xBig)
+
+			err = ctx.ScopeManager.AssignVariable("slope", slopeBig)
+			if err != nil {
+				return err
+			}
+
+			err = ctx.ScopeManager.AssignVariable("x0", x0Big)
+			if err != nil {
+				return err
+			}
+
+			err = ctx.ScopeManager.AssignVariable("x1", x1Big)
+			if err != nil {
+				return err
+			}
+
+			err = ctx.ScopeManager.AssignVariable("y0", y0Big)
+			if err != nil {
+				return err
+			}
+
+			err = ctx.ScopeManager.AssignVariable("new_x", new_xBig)
+			if err != nil {
```

Comment:
Should we add something like `AssignVariables` for the ScopeManager? You would just need to pass a map with all the variables you want to assign and check for error just once after

Reply:
I think its a good idea. Added.